spec/solver_spec.rb

```diff
@@ -0,0 +1,61 @@
+require_relative './../solver'
+
+describe 'Solver' do
+  describe 'When testing factorial' do
+    it 'Factorial test' do
+      number = 5
+      solver = Solver.new
+      factorial = solver.factorial(number)
+      expect(factorial).to eq(120)
+    end
```

Comment:
- [ ] for the factorial method you should also write tests for negative numbers and 0.